### ISSUE TITLE
RPM for RedHat 8

### ISSUE THREAD

**Issue by its404** [certbot/certbot#7202]
If you're having trouble using Certbot and aren't sure you've found a bug or
request for a new feature, please first try asking for help at
https://community.letsencrypt.org/. There is a much larger community there of
people familiar with the project who will be able to more quickly answer your
questions.

## My operating system is (include version):
RedHat 8

## Question:
There is no installation package available at RedHat 8 yet, is there a plan for making installation package for RedHat 8? Thanks.

**Comment by bmw** [comment]
I think this a question for you @elyscape. I assume it's blocked on the release of EPEL 8?

**Comment by elyscape** [comment]
Correct. As far as I'm aware, EPEL 8 does not yet exist. Once it does, I'll get the certbot RPMs into it.

**Comment by haidarvm** [comment]
Does Anyone try it & how to install it manual ?
I got this error when I run  ./certbot-auto --debug --nginx 
```
Traceback (most recent call last):
  File "<stdin>", line 27, in <module>
  File "<stdin>", line 19, in create_venv
  File "/usr/lib64/python2.7/subprocess.py", line 185, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib64/python2.7/subprocess.py", line 172, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib64/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/usr/lib64/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```
Anyone knows how to run it correctly with Red Hat Enterprise Linux 8 ?

**Comment by bmw** [comment]
@haidarvm, can you create a new issue to track certbot-auto support for RHEL 8 and include the output of `./certbot-auto --debug --nginx --verbose` in the post? `--verbose` may give us a little more insight into what's going wrong.

**Comment by haidarvm** [comment]
`sudo ./certbot-auto  --debug --nginx --verbose
Bootstrapping dependencies for RedHat-based OSes... (you can skip this with --no-bootstrap)
dnf is /bin/dnf
dnf is hashed (/bin/dnf)
Updating Subscription Management repositories.
Last metadata expiration check: 2:03:06 ago on Jum 12 Jul 2019 12:03:04  +08.
Package gcc-8.2.1-3.5.el8.x86_64 is already installed.
Package augeas-libs-1.10.1-8.el8.x86_64 is already installed.
Package openssl-1:1.1.1-8.el8.x86_64 is already installed.
Package openssl-devel-1:1.1.1-8.el8.x86_64 is already installed.
Package libffi-devel-3.1-18.el8.x86_64 is already installed.
Package redhat-rpm-config-116-1.el8.noarch is already installed.
Package ca-certificates-2018.2.24-6.el8.noarch is already installed.
Package python2-libs-2.7.15-22.module+el8.0.0+2961+596d0223.x86_64 is already installed.
Package python2-setuptools-39.0.1-11.module+el8.0.0+2961+596d0223.noarch is already installed.
Package python2-devel-2.7.15-22.module+el8.0.0+2961+596d0223.x86_64 is already installed.
Package python2-virtualenv-15.1.0-18.module+el8.0.0+2961+596d0223.noarch is already installed.
Package python2-tools-2.7.15-22.module+el8.0.0+2961+596d0223.x86_64 is already installed.
Package python2-pip-9.0.3-13.module+el8.0.0+2961+596d0223.noarch is already installed.
Dependencies resolved.
Nothing to do.
Complete!
./certbot-auto has insecure permissions!
To learn how to fix them, visit https://community.letsencrypt.org/t/certbot-auto-deployment-best-practices/91979/
Creating virtual environment...
Traceback (most recent call last):
  File "<stdin>", line 27, in <module>
  File "<stdin>", line 19, in create_venv
  File "/usr/lib64/python2.7/subprocess.py", line 185, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib64/python2.7/subprocess.py", line 172, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib64/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/usr/lib64/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
`
@bmw thanks for support, I always got the same error either with sudo or user
Please, I really rely on certbot for my trusted HTTPS Certification. I can't show it to client without https

**Comment by its404** [comment]
The instruction for RHEL 8 is available on website now, can follow it to install certbot and use it. Thanks.

**Comment by jgilbert35** [comment]
EPEL 8 has been released. Can the certbot RPMs be added to it?

**Comment by elyscape** [comment]
I’m going to investigate what work is necessary to make this happen, but be advised that it will likely take a while, as all the dependencies will also need to be added to EPEL8 first.

**Comment by bmw** [comment]
The Certbot package was added to EPEL 8 recently, but its plugins, especially the Apache and Nginx, plugins still need to be packaged.

cc @FelixSchwarz

**Comment by FelixSchwarz** [comment]
@bmw: Unfortunately I'm not what the intended process for Fedora's certbot packages is. I'm using only plain certbot so I was able to work on that.

@elyscape You requested EPEL8 branches for the various `certbot-…` packages. AFAIK you have some pretty extensive scripts to automate updates for all the plugins. Do you have any plans to make the plugins available for EPEL 8, too?

**Comment by elyscape** [comment]
@FelixSchwarz Yes. I need to ensure that all the dependencies are packaged (and several likely aren’t), but then I’ll be bringing them forward.

**Comment by bmw** [comment]
Thanks. I'm mainly interested in the Apache and Nginx plugins because then we can update our website to tell CentOS/RHEL 8 users to stop using certbot-auto which we're planning on deprecating soon.

Taking a quick look, I don't believe either of those plugins have dependencies that need to be packaged.

**Comment by systemcrash** [comment]
Does this ticket regard certbot only or also the certbot modules (which are based on Python 2.x)?

**Comment by bmw** [comment]
It also involves the Certbot modules/plugins, however, they should be packaged for Python 3 rather than Python 2. Certbot has supported Python 3 for years and is planning on dropping Python 2 support early next year.

**Comment by FelixSchwarz** [comment]
@elyscape Is there a bugzilla ticket so I can track progress/help with some of the plugins?  I understand if you lack the time to drive this forward. In that case we should make sure that others can understand the process of updating all the plugins in an efficient manner.

**Comment by bmw** [comment]
Some Bugzilla tickets for this have recently been created:

* https://bugzilla.redhat.com/show_bug.cgi?id=1795666
* https://bugzilla.redhat.com/show_bug.cgi?id=1797126
* https://bugzilla.redhat.com/show_bug.cgi?id=1786479

While I would love to see all of Certbot's plugins packaged, I would strongly recommend prioritizing the Apache and Nginx plugins as they are multiple orders of magnitude more popular than our DNS plugins. See https://gist.github.com/bmw/0ef8f41d6dc70ec76d6545b9af3a9c78 if you're curious about the stats there.

**Comment by FelixSchwarz** [comment]
Just wanted to mention that I just pushed certbot 1.3 + plugins to EPEL8 stable. It may take 1-2 days until the RPMs reach all mirrors though. The only exception is the digitalocean dns plugin as I need to upgrade the digitalocean package in EPEL8 but I don't have enough permissions to do so (ping @elyscape).

As I now have the scripts in place I hope we can push new versions to Fedora+EPEL much faster. I'd love to get some help from Fedora packagers for moving certbot in EPEL 7 to Python 3 (see https://bugzilla.redhat.com/show_bug.cgi?id=1797129 + dependent bugs).

**Comment by bmw** [comment]
Thanks so much for working on this @FelixSchwarz! I really appreciate it.

Unfortunately, no one on the Certbot team at EFF has any RPM packaging experience, but let me know if you'd like me to post messages to places like https://community.letsencrypt.org and various IRC channels making a public call for people with this experience to help.

**Comment by FelixSchwarz** [comment]
> Unfortunately, no one on the Certbot team at EFF has any RPM packaging experience, but let me know if you'd like me to post messages to places like https://community.letsencrypt.org and various IRC channels making a public call for people with this experience to help.

Actually my comment was not directed at existing certbot developers. I suspect that some subscribers to this ticket are also Fedora packagers interested in certbot so I mentioned some Fedora tasks :-)

Btw: If someone tries to install EPEL's certbot package and lacks the python-mock package, please see my [comment in bugzilla about the PowerTools repo](https://bugzilla.redhat.com/show_bug.cgi?id=1821439#c1)

**Comment by FelixSchwarz** [comment]
Just wanted to mention that EPEL 8 has certbot 1.5 now which has no `mock` dependency anymore so installation should be trivial. I think this issue can be closed as the certbot core team is unlikely to publish "official" `certbot` rpms given their time constraints.

**Comment by bmw** [comment]
You're right. Thanks a lot for creating those packages and for the updates here.
